### PR TITLE
feat: add condition for min node version

### PR DESCRIPTION
**Pull request by szymonrybczak**
Summary:
---------

Closes https://github.com/react-native-community/cli/issues/1768

Test Plan:
----------

1. Clone the repository and do all the required steps from the [Contributing guide](https://github.com/react-native-community/cli/blob/main/CONTRIBUTING.md)
2. Download old version of `node` e.g. `v12.22.12`
3. Run `node packages/cli/build/bin.js start` inside repo and, you should get an error `You need at least Node 14 to run CLI.`

### CODE REVIEW

File: packages/cli/src/bin.ts [hunk 1]
```diff
@@ -1,7 +1,15 @@
 #!/usr/bin/env node
 
 import './tools/gracefulifyFs';
+import semver from 'semver';
+import execa from 'execa';
+import chalk from 'chalk';
 
-import {run} from './';
+const nodeVersion = execa.sync('node', ['--version'], {}).stdout;
```
Comment (thymikee) [inline packages/cli/src/bin.ts:8]: you can get node's version from `process.version`

**Review by thymikee — CHANGES_REQUESTED** [review]
I think it's weird that npx doesn't respect `engines` entry in package.json. Until it does, I'm good with this check, just let's make it less of a perf impact

File: packages/cli/src/bin.ts [hunk 2]
```diff
@@ -1,7 +1,12 @@
 #!/usr/bin/env node
 
 import './tools/gracefulifyFs';
+import semver from 'semver';
+import chalk from 'chalk';
 
-import {run} from './';
-
-run();
+if (semver.satisfies(process.version, '>=14')) {
```
Comment (adamTrz) [inline packages/cli/src/bin.ts:7]: Maybe worth taking it from `versionRanges.ts` file?  🤔 
https://github.com/react-native-community/cli/blob/b06993e72dccc2658551b76a90b9f063e9cb8731/packages/cli-doctor/src/tools/versionRanges.ts#L3

Comment (thymikee) [reply packages/cli/src/bin.ts:7]: even better

File: packages/cli/src/bin.ts [hunk 3]
```diff
@@ -1,7 +1,12 @@
...
+  const {run} = require('./');
+  run();
+} else {
+  console.error(`${chalk.red('You need at least Node 14 to run CLI.')}`);
```
Comment (thymikee) [inline packages/cli/src/bin.ts:11]: How about something like this:
> React Native needs Node.js >=14. You're currently on version 12.3.4. Please upgrade Node.js to a supported version and try again.
```suggestion
  console.error(`${chalk.red('React Native needs Node.js ${nodeVersionRange}. You're currently on version ${process.version}. Please upgrade Node.js to a supported version and try again.')}`);
```

File: packages/cli/src/bin.ts (hunk 2 above)
Comment (szymonrybczak) [reply packages/cli/src/bin.ts]: fixed ✅

File: packages/cli/src/bin.ts (hunk 3 above)
Comment (szymonrybczak) [reply packages/cli/src/bin.ts]: Looks good. Updated 😄

**Review by thymikee — APPROVED** [review]
Can you confirm this still works after extra imports? Ideally we'd have a test for that but It may be tricky to come up with a good one. Mocking process.version will not take module resolution into account and otherwise we'd need to run this through node@12

**Comment by szymonrybczak** [comment]
Yeah, I tested that with node@12 and it still works 😄

**Review by adamTrz — APPROVED** [review]

**Pull request merged**